crates/task-impls/src/events.rs

```diff
@@ -94,17 +94,19 @@ pub enum HotShotEvent<TYPES: NodeType, I: NodeImplementation<TYPES>> {
     TransactionsRecv(Vec<TYPES::Transaction>),
     /// Send transactions to the network
     TransactionSend(TYPES::Transaction, TYPES::SignatureKey),
+    /// Transactions have been sequenced
+    TransactionsSequenced(
+        TYPES::BlockPayload,
+        <TYPES::BlockPayload as BlockPayload>::Metadata,
+        TYPES::Time,
+    ),
```

Comment:
Doesn't the `BlockPayload` type currently require the vid commitment?  If so, then I think we need to change this `BlockPayload` type here to something else, like a type that just contains the transactions.

Reply:
Ah yes, looks like it _does_ calculate the vid commitment at that step. Let me fix it

Reply:
`BlockPayload` should not need the VID commitment.  Quoting from [Jeb's zulip post](https://github.com/EspressoSystems/HotShot/pull/2077#discussion_r1397848207):

I think a minimal interface required to completely solve this problem would be something like

```rust
trait BlockPayload {
  type Transaction;
  type Metadata; // Data created during block building which feeds into the block header, e.g. namespace table

  fn build(transactions: impl IntoIterator<Item = Transaction>) -> (Self, Self::Metadata);
  fn encode(&self) -> Vec<u8>;
  fn decode(bytes: &[u8]) -> Self;
}

trait BlockHeader<Metadata> {
  fn new(payload_commitment: VidCommitment, metadata: Self::Metadata, parent_leaf: Leaf) -> Self;
}
```

Then the workflow would be like this:
* Transactions task uses `build` to create a block payload and namespace table
* Transactions task uses `encode` to flatten block payload into bytes, computes VID commitment
* Leader sends flat bytes *and* metadata to DA committee members
* Committee members compute VID commitment over flat bytes, sign hash of VID commitment and metadata
* Leader uses `BlockHeader::new` to construct block header (and thus proposal) from VID commitment and metadata
* Leader sends proposal including header and DAC to replicas
* Replicas extract metadata and VID commitment from header and check against the signatures in DAC

Reply:
Relevant discussion: https://github.com/EspressoSystems/HotShot/pull/2045#discussion_r1389986724 (Sorry I'm in a rush and unsure of the final design atm.)

Reply:
There was debate in the above linked thread on whether to put `Metadata` in the payload or header. But we can leave it as-is for now. The important part is that the VID commitment goes with the parent leaf in the header, *not in the payload*.

Reply:
I think this should also include the encoded transactions, which with the metadata the leader can use to get the block payload and the commitment.

Reply:
Okay, so I think the plan is now as follows:

Since @shenkeyao is removing commit(), the changes for _this_ PR will just be to include the encode transactions in `TransactionsSequenced`.